src/code/z_player_lib.c

```diff
@@ -2045,13 +2609,195 @@ void Player_DrawGetItem(PlayState* play, Player* player) {
+            return;
+        }
+
+        Matrix_Push();
+        Matrix_Scale(player->unk_AF0[0].x, player->unk_AF0[0].y, player->unk_AF0[0].z, MTXMODE_APPLY);
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, D_801C0ABC[arg2]);
+
+        Matrix_MultVec3f(&D_801C0AC4[arg2], &sp58);
+        Matrix_MultVec3f(&D_801C0ADC[arg2], &sp4C);
+
+        if (func_80126440(play, NULL, &player->meleeWeaponInfo[arg2], &sp58, &sp4C) &&
+            (player->stateFlags1 & PLAYER_STATE1_8000000)) {
+            EffectBlure_AddVertex(Effect_GetByIndex(player->meleeWeaponEffectIndex[arg2]),
+                                  &player->meleeWeaponInfo[arg2].tip, &player->meleeWeaponInfo[arg2].base);
+        }
+        Matrix_Pop();
+
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+}
+
+s32 func_801271B0(PlayState* play, Player* player, s32 arg2) {
+    if (player->transformation == PLAYER_FORM_DEKU) {
+        if (((player->skelAnime.animation == &gPlayerAnim_pn_kakku)) ||
+            (player->skelAnime.animation == &gPlayerAnim_pn_kakkufinish) ||
+            (player->skelAnime.animation == &gPlayerAnim_pn_rakkafinish) ||
+            (player->skelAnime.animation == &gPlayerAnim_pn_batabata)) {
+            struct_80124618** sp3C = D_801C0AF4;
+
+            OPEN_DISPS(play->state.gfxCtx);
+
+            if (player->skelAnime.animation == &gPlayerAnim_pn_kakkufinish) {
+                sp3C = D_801C0AFC;
+            } else if (player->skelAnime.animation == &gPlayerAnim_pn_batabata) {
+                sp3C = D_801C0B0C;
+            } else if (player->skelAnime.animation == &gPlayerAnim_pn_rakkafinish) {
+                sp3C = D_801C0B04;
+            }
+
+            Matrix_Push();
+            Matrix_Translate(0.0f, 150.0f, 0.0f, MTXMODE_APPLY);
+            func_80124618(sp3C[0], player->skelAnime.curFrame, &player->unk_AF0[1]);
+            Matrix_Scale(player->unk_AF0[1].x, player->unk_AF0[1].y, player->unk_AF0[1].z, 1);
+
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+            gSPDisplayList(POLY_OPA_DISP++, D_801C0B14[arg2]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80126B8C.s")
+            Matrix_Translate(2150.0f, 0.0f, 0.0f, MTXMODE_APPLY);
+            Matrix_RotateXS(player->unk_B8A, MTXMODE_APPLY);
+            func_80124618(sp3C[1], player->skelAnime.curFrame, &player->unk_AF0[1]);
+            Matrix_Scale(player->unk_AF0[1].x, player->unk_AF0[1].y, player->unk_AF0[1].z, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80126BD0.s")
+            gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_801271B0.s")
+            // Close flower / Open flower
```

Comment:
Which flower? You might as well rename the DLs anyway...

Reply:
![image](https://user-images.githubusercontent.com/7416381/232899931-f9d6dc83-09f0-460e-8701-3a690a782ee5.png)
![image](https://user-images.githubusercontent.com/7416381/232899992-3f2e9314-82b7-4d47-a1ac-b7ff1aa653b5.png)

I'm not sure how to name this dl, `gLinkDekuClosedFlowerDL`/`gLinkDekuOpenFlowerDL`?

Reply:
That sounds fine to me, at least.